media-video/mjpg-streamer/mjpg-streamer-0_pre20200524-r1.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit eutils systemd
+
+DESCRIPTION="MJPG-streamer takes JPGs from Linux-UVC compatible webcams"
+HOMEPAGE="https://github.com/jacksonliam/mjpg-streamer"
+EGIT_COMMIT="5a6e0a2db163e6ae9461552b59079870d0959340"
+SRC_URI="https://github.com/jacksonliam/${PN}/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+INPUT_PLUGINS="input-testpicture input-control input-file input-uvc input-http input-opencv input-ptp2"
+OUTPUT_PLUGINS="output-file output-udp output-http output-autofocus output-rtsp output-viewer output-zmqserver"
+IUSE_PLUGINS="${INPUT_PLUGINS} ${OUTPUT_PLUGINS}"
+IUSE="input-testpicture input-control +input-file input-uvc input-http input-opencv input-ptp2
+	output-file	output-udp +output-http output-autofocus output-rtsp output-viewer output-zmqserver
+	www http-management wxp-compat"
+REQUIRED_USE="|| ( ${INPUT_PLUGINS} )
+	|| ( ${OUTPUT_PLUGINS} )"
+
+RDEPEND="virtual/jpeg
+	input-uvc? ( media-libs/libv4l )
+	input-opencv? ( media-libs/opencv )
+	output-zmqserver? ( dev-libs/protobuf-c net-libs/zeromq )
+	input-ptp2? ( media-libs/libgphoto2 )"
+DEPEND="${RDEPEND}
+	input-testpicture? ( media-gfx/imagemagick )"
+
+S="${WORKDIR}/${PN}-${EGIT_COMMIT}/mjpg-streamer-experimental"
+
+src_prepare() {
+	sed -i -e "s|.*RPATH.*||g" CMakeLists.txt
+	use wxp-compat && sed -i -e \
+		's|^add_feature_option(WXP_COMPAT "Enable compatibility with WebcamXP" OFF)|add_feature_option(WXP_COMPAT "Enable compatibility with WebcamXP" ON)|g' CMakeLists.txt
+
+	local flag switch
+	for flag in ${IUSE_PLUGINS}; do
+		use ${flag} && switch='' || switch='#'
+		flag=${flag/input-/input_}
+		flag=${flag/output-/output_}
```

Comment:
Honestly I think this plugin management is just overcomplicating this ebuild. You can combine some of the flags, and enable everything that doesn't need external dependencies by default.

Reply:
The thing is, that the outcome of build is main binary mjpg_streamer and plugin binaries. And not always all the plugin binaries are needed to be built/loaded. That's why I've given user ability to choose. Other flags are conditionals for compatibilities with weird devices or bug overcoming or simple optional features.

Reply:
Now I can see that unfortunately this project is dropping support for media-libs/opencv-4.X.X and only supporting media-libs/opencv-3.x which is sadly not in portage anymore, so I will drop this use flag also. Others plugins and use flags to keep?

Reply:
Are there any plugins we can enable unconditionally? (Those not dragging in extra deps?)

And combine those which are logically similar.

The point is that exposing every single possible combination upstream isn't necessarily what we want to do in the ebuild -- it makes testing really hard.

Reply:
Yes, by default only USE="input-file* output-http" is enabled which is not pulling new deps. Only by enabling more/other plugins, it pulls more deps.

Reply:
OK. But there are no others with no external deps which we could enable?

Reply:
No, better set minimal requirements and let user's decide what to pull in. Currently I see#  qdepends mjpg-streamer
media-video/mjpg-streamer-0_pre20200524-r1: virtual/pkgconfig media-libs/libv4l virtual/jpeg